fails with:

##Failure Location unknown## : Error
Test name: HSQLDBTest::testEmptyDBConnection
An uncaught exception of type com.sun.star.sdbc.SQLException
- General error: java.lang.UnsatisfiedLinkError: com.sun.star.sdbcx.comp.hsqldb
.StorageFileAccess.isStreamElement(Ljava/lang/String;Ljava/lang/String;)Z

which suggest JNI linkage problem but it doesn't help to rebuild hsqldb
with openjdk-7...

diff --git a/dbaccess/Module_dbaccess.mk b/dbaccess/Module_dbaccess.mk
index e9eb146..0c9aa49 100644
--- a/dbaccess/Module_dbaccess.mk
+++ b/dbaccess/Module_dbaccess.mk
@@ -42,10 +42,12 @@ $(eval $(call gb_Module_add_check_targets,dbaccess,\
 ))
 
 ifeq ($(ENABLE_JAVA),TRUE)
+ifneq ($(SYSTEM_HSQLDB),YES)
 $(eval $(call gb_Module_add_check_targets,dbaccess,\
     CppunitTest_dbaccess_hsqldb_test \
 ))
 endif
+endif
 
 # This runs a suite of peformance tests on embedded firebird and HSQLDB.
 # Instructions on running the test can be found in qa/unit/embeddedb_performancetest
